app/models.py

```diff
@@ -1551,17 +1558,26 @@ def serialize(self):
     AuditEvent.created_at,
 )
 
-# Index for type + acknowledged audit event query count - for admin
-# app "Service updates" requests: while the page of audit events
-# will rely on the created_at index scan and is relatively quick,
-# count(*) query executed by `.paginate` is very slow without a
-# dedicated index for objects with multiple events
+
+# DEPRECATED - remove in a migration once supplier update admin app has been updated
 db.Index(
     'idx_audit_events_type_acknowledged',
     AuditEvent.type,
     AuditEvent.acknowledged,
 )
 
+# partial index for supplier_update-monitoring admin views. specifically this
+# ordering should allow the index to be used for earliest_for_each_object's
+# ordering-followed-by-DISTINCT ON
+db.Index(
+    'idx_audit_events_created_at_per_obj_partial',
+    AuditEvent.object_type,
+    AuditEvent.object_id,
+    AuditEvent.created_at,
+    AuditEvent.id,
+    postgresql_where=sql_and(AuditEvent.acknowledged == sql_false(), AuditEvent.type == "supplier_update"),
```

Comment:
I'm going to rebase this out so it's like it never happened and you'll have a hard time proving that it did.